truffle-config.js

```diff
@@ -1,18 +1,126 @@
-/*
- * NB: since truffle-hdwallet-provider 0.0.5 you must wrap HDWallet providers in a 
- * function when declaring them. Failure to do so will cause commands to hang. ex:
- * ```
- * mainnet: {
- *     provider: function() { 
- *       return new HDWalletProvider(mnemonic, 'https://mainnet.infura.io/<infura-key>') 
- *     },
- *     network_id: '1',
- *     gas: 4500000,
- *     gasPrice: 10000000000,
- *   },
+/**
+ * Use this file to configure your truffle project. It's seeded with some
+ * common settings for different networks and features like migrations,
+ * compilation and testing. Uncomment the one's you need or modify
+ * them to suit your project as necessary.
+ *
+ * This file is named `truffle-config` for Windows users because `truffle.js`
+ * conflicts with the `truffle` command on that platform. (You can also
+ * use this file on Linux and Mac.)
```

Comment:
Should we switch to only one file for Truffle v5? (https://github.com/trufflesuite/truffle/issues/895)

Reply:
Yes. But I'm too bewildered by that discussion to know what the right one is. Is it `truffle-config.js`?

Reply:
I was trying to remain neutral but I do think `truffle-config.js` is the correct one. I know @bmmpxf prefers `truffle.js` but I think it's better just to avoid needing the extra Windows configuration step. Or axic's suggestion of `truffle.config.js`, but I'm not sure the value here. `truffle-config.js` doesn't require a code change.

Note: I think we shouldn't stop `truffle.js` from working; instead just stop including it in the boxen.

Reply:
👍 Yes sounds right to me too. . . it's the smallest change. Might need to grep through the docs and update there as well. 

In any case I'll remove `truffle.js` from this and get rid of the windows talk altogether.